src/common/cockpitchannel.c

```diff
@@ -367,9 +373,9 @@ cockpit_channel_actual_send (CockpitChannel *self,
           json_object_unref (ping);
         }
 
-      /* If we've sent more than the window, apply back pressure */
       priv->out_sequence = out_sequence;
-      if (priv->out_sequence > priv->out_window)
+
+      if (under_pressure)
         {
           g_debug ("%s: sent too much data without acknowledgement, emitting back pressure until %"
                    G_GINT64_FORMAT, priv->id, priv->out_window);
```

Comment:
That means this emit_pressure call here has also changed from level triggered to edge triggered. Is that in line with all the other uses of this function call? Worth doing a check.

Reply:
> src/bridge/cockpitpacketchannel.c:    cockpit_flow_emit_pressure (COCKPIT_FLOW (self), FALSE);
> src/bridge/cockpitpacketchannel.c:    cockpit_flow_emit_pressure (COCKPIT_FLOW (self), TRUE);

Edge triggered. ^^

> src/bridge/cockpitstream.c:            cockpit_flow_emit_pressure (COCKPIT_FLOW (self), FALSE);
> src/bridge/cockpitstream.c:    cockpit_flow_emit_pressure (COCKPIT_FLOW (self), TRUE);

Edge triggered ^^

> src/common/cockpitchannel.c:        cockpit_flow_emit_pressure (COCKPIT_FLOW (self), FALSE);

This other call in CockpitChannel is edge triggered. But may have an off by one on line 248. I need to double check this.

> src/common/cockpitchannel.c:          cockpit_flow_emit_pressure (COCKPIT_FLOW (self), TRUE);

This is the one we're changing.

> src/common/cockpitpipe.c:    cockpit_flow_emit_pressure (COCKPIT_FLOW (self), FALSE);
> src/common/cockpitpipe.c:    cockpit_flow_emit_pressure (COCKPIT_FLOW (self), TRUE);

Edge triggered ^^

src/common/cockpitwebresponse.c:        cockpit_flow_emit_pressure (COCKPIT_FLOW (self), FALSE);
src/common/cockpitwebresponse.c:    cockpit_flow_emit_pressure (COCKPIT_FLOW (self), TRUE);

Edge triggered ^^

src/websocket/websocketconnection.c:    cockpit_flow_emit_pressure (COCKPIT_FLOW (self), FALSE);
src/websocket/websocketconnection.c:    cockpit_flow_emit_pressure (COCKPIT_FLOW (self), TRUE);

Edge triggered ^^

Reply:
As we just figured out, all other pressure calls use edge triggering already, so that gets cockpitchannel.c in line with the others.